TMT Method Chapter

moorepants opened the pull request.

moorepants commented:
@Peter230655 the basics draft is in place for this chapter. But I can't find the issue for why I don't get the same results as before. Maybe you can see it. I'm not sure if I have an error here or an error in the prior example in the simulation chapter.

moorepants commented:
The first two entries of the `gd` term I have in this chapter do not match the first two terms in the prior introduction of this problem where I solve it with Kane's method. The symbolics and numerics can be viewed here for the prior version:

https://moorepants.github.io/learn-multibody-dynamics/simulation.html#numerical-evaluation

`gd` from this TMT chapter:

```
array([[-4.49020839],
        [-0.02484139],
        [-1.1112791 ]])
```

`gd` from the Kane chapter(s):

```
array([[-4.48963535],
        [-0.02486744],
        [-1.1112791 ]])
```

The first entry of the TMT method `gq` has:

$$
\frac{l^{2} m \sin{\left(2 q_{2} \right)} \dot{q}_{1} \dot{q}_{2}}{24}
$$

where as the original has the same term by divided by 12 instead of 24. This term comes from the "I x w * I" part of Euler's equations, i.e. an inertia times omega1*omega2 type terms. So I think this is the best hint to the error.

I either have an error in my Kane's derivation or in the TMT but I have not been able to find out what I'm doing incorrectly. The results should match.

Peter230655 commented:
Dear Jason,

Would it help, if I tried to set up this pendulum using KanesMethod() ?
I have modelled many pendulums, so it should not be too much trouble doing
it.  :-))

Peter

Am Mi., 8. Juni 2022 um 07:06 Uhr schrieb Jason K. Moore <
***@***.***>:

> The first two entries of the gd term I have in this chapter do not match
> the first two terms in the prior introduction of this problem where I solve
> it with Kane's method. The symbolics and numerics can be viewed here for
> the prior version:
>
>
> https://moorepants.github.io/learn-multibody-dynamics/simulation.html#numerical-evaluation
>
> gd from this TMT chapter:
>
> array([[-4.49020839],
>         [-0.02484139],
>         [-1.1112791 ]])
>
> gd from the Kane chapter(s):
>
> array([[-4.48963535],
>         [-0.02486744],
>         [-1.1112791 ]])
>
> The first entry of the TMT method gq has:
>
> $$ \frac{l^{2} m \sin{\left(2 q_{2} \right)} \dot{q}*{1} \dot{q}*{2}}{24}
> $$
>
> where as the original has the same term by divided by 12 instead of 24.
> This term comes from the "I x w * I" part of Euler's equations, i.e. an
> inertia times omega1*omega2 type terms. So I think this is the best hint to
> the error.
>
> I either have an error in my Kane's derivation or in the TMT but I have
> not been able to find out what I'm doing incorrectly. The results should
> match.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/moorepants/learn-multibody-dynamics/pull/80#issuecomment-1149465923>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQURRJ43E2J7O2UTRWNDVOAS4VANCNFSM5YAII5XA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

moorepants commented:
Peter,

Yes, if you solve it interdependently, then I would know if my Kane or my TMT implementation is incorrect. That would help to narrow down to where to look for the bug.

Jason

Peter230655 commented:
Working on it already! :-))

On Wed 8. Jun 2022 at 14:19, Jason K. Moore ***@***.***>
wrote:

> Peter,
>
> Yes, if you solve it interdependently, then I would know if my Kane or my
> TMT implementation is incorrect. That would help to narrow down to where to
> look for the bug.
>
> Jason
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/moorepants/learn-multibody-dynamics/pull/80#issuecomment-1149639586>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUSTLTFX3UNZKHDPMUTVOBNCDANCNFSM5YAII5XA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
Best regards,

Peter Stahlecker

moorepants commented:
Peter sent me his Kane solution that used KanesMethod() and it gives the same answer as my Kane solution, so my error must be in the TMT page.

moorepants commented:
I found the error. When forming gbar in the TMT equations I was doing M*vbar.diff(t), effectively but I have my inertia's expressed in terms of q (they change with time) so you have to do (M*vbar).diff(t)!

moorepants commented:
@jtheinen  and @Peter230655 I found the bug and fixed the code and updated the text. I'm going to publish this because I'm already late sending it to the students. But if you want to review my text, feel free to do so and I can edit after the fact. Thanks for all the help!

moorepants commented:
BTW, this is the last chapter for the purposes of the class! Hooray! Nice accomplishment. You two have been a great help. I very much appreciate it :)

Peter230655 commented:
Dear Jason,

My knowledge of Python and of sympy is not that great, so I do not really
understand:

What is the difference between:

*Mvbar.diff(t)*

and

*(Mvbar).diff(t)*

Thanks!

Peter

On Thu 9. Jun 2022 at 11:00 Jason K. Moore ***@***.***> wrote:

> I found the error. When forming gbar in the TMT equations I was doing M*vbar.diff(t),
> effectively but I have my inertia's expressed in terms of q (they change
> with time) so you have to do (M*vbar).diff(t)!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/moorepants/learn-multibody-dynamics/pull/80#issuecomment-1150686880>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQURNHVHUGI2C7FYYWLDVOF6OVANCNFSM5YAII5XA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
Best regards,

Peter Stahlecker

moorepants commented:
Peter,

The difference is mathematical. $\mathbf{M}\frac{d\bar{v}}{dt}$ is different than $\frac{d\mathbf{M}\bar{v}}{dt}$ if $\mathbf{M}$ is a function of time, which it is in my presentation where I express the inertias in the N reference frame, making them functions of the coordinates.

Peter230655 commented:
Dear Jason,

Clear, I missed the multiplication, did not realize that Mv.diff(t) meant
M*v.diff(t).
Thanks!

Peter

On Fri 10. Jun 2022 at 10:30, Jason K. Moore ***@***.***>
wrote:

> Peter,
>
> The difference is mathematical. $\mathbf{M}\frac{d\bar{v}}{dt}$ is
> different than $\frac{d\mathbf{M}\bar{v}}{dt}$ if $\mathbf{M}$ is a
> function of time, which it is in my presentation where I express the
> inertias in the N reference frame, making them functions of the coordinates.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/moorepants/learn-multibody-dynamics/pull/80#issuecomment-1151942526>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUTOKVHEVA7U5TYIR7TVOLDVLANCNFSM5YAII5XA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
Best regards,

Peter Stahlecker

Peter230655 commented:
Dear Jason,

You wrote to sympy:



*Once I get the PDF to build, I think it will amount to about 300 pages or
so. It is fairly complete and can mostly standalone*

Once you have this PDF, can you send me a copy, or would whatever
proprietary rights prevent you from doing so?

Thanks!

Peter

On Fri 10. Jun 2022 at 20:47 Peter Stahlecker ***@***.***>
wrote:

> *Unconstrained Equations of Motion with the TMT Method*
>
> 1. Eq(180)
>
> You write:
>
> 𝐓=𝐉𝑣¯,𝑞¯∈ℝ6𝜈×𝑛where𝑣¯=𝐓𝑢¯
>
> 1.1.
> I believe, n is the number of generalized coordinates for the ny bodies,
> as you mention below, below eq(185), but would it be a good idea to mention
> it here, where you first introduce it?
>
> 1.2.
> I assume that J_subscript(v_bar), subscript(q_bar) means
> v_bar.Jacobian(q_bar)
>
> If I am right, should it not be J_subscript(v_bar), subscript(*u_bar*) ?
>
> At least, it seems that this is what you are doing here:
>
> T = v.jacobian(u)
>
>
> 2. Eq(186, 187, 188, 189)
>
> Are these equations not the ‚heart‘ of the TMT Method?
>
> 3.
>
> You write:
>
> R_Bo = m*g*N.x + kl*q3*B.yR_Q = m/4*g*N.x - kl*q3*B.yT_A = -kt*q1*N.z + kt*q2*A.x
>
>
> Of course, you have mentioned it before, but would it make sense to
> mention that in
>
> R_Bo = m*g*N.x *+ kl*q3*B.y*
>
> the bold part is the ‚reactio force‘ on Bo from the particle Q?
>
> ( I missed this when I tried to model this pendulum with KanesMethod(). It
> should be obvious, but I failed to see it)
>
>
> 4.
> For my further education:
>
> Below eq(189), you write that these equations are equivalent to Kane‘s
> Equations.
> Is there an advantage to the TMT method?
> It should be, as it was developed 35 years later.
>
>
>
>
> On Fri 10. Jun 2022 at 18:25 Peter Stahlecker ***@***.***>
> wrote:
>
>> Dear Jason,
>>
>> Clear, I missed the multiplication, did not realize that Mv.diff(t) meant
>> M*v.diff(t).
>> Thanks!
>>
>> Peter
>>
>> On Fri 10. Jun 2022 at 10:30, Jason K. Moore ***@***.***>
>> wrote:
>>
>>> Peter,
>>>
>>> The difference is mathematical. $\mathbf{M}\frac{d\bar{v}}{dt}$ is
>>> different than $\frac{d\mathbf{M}\bar{v}}{dt}$ if $\mathbf{M}$ is a
>>> function of time, which it is in my presentation where I express the
>>> inertias in the N reference frame, making them functions of the coordinates.
>>>
>>> —
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/moorepants/learn-multibody-dynamics/pull/80#issuecomment-1151942526>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/AT5MQUTOKVHEVA7U5TYIR7TVOLDVLANCNFSM5YAII5XA>
>>> .
>>> You are receiving this because you were mentioned.Message ID:
>>> ***@***.***>
>>>
>> --
>> Best regards,
>>
>> Peter Stahlecker
>>
> --
> Best regards,
>
> Peter Stahlecker
>
-- 
Best regards,

Peter Stahlecker

moorepants commented:
Peter,

Yes, for sure. It will probably be sometime later this summer. I have to
fix a bunch of things for the PDF to build correctly.

Jason
moorepants.info
+01 530-601-9791


On Sat, Jun 11, 2022 at 4:55 AM Peter230655 ***@***.***>
wrote:

> Dear Jason,
>
> You wrote to sympy:
>
>
>
> *Once I get the PDF to build, I think it will amount to about 300 pages or
> so. It is fairly complete and can mostly standalone*
>
> Once you have this PDF, can you send me a copy, or would whatever
> proprietary rights prevent you from doing so?
>
> Thanks!
>
> Peter
>
> On Fri 10. Jun 2022 at 20:47 Peter Stahlecker ***@***.***>
> wrote:
>
> > *Unconstrained Equations of Motion with the TMT Method*
> >
> > 1. Eq(180)
> >
> > You write:
> >
> > 𝐓=𝐉𝑣¯,𝑞¯∈ℝ6𝜈×𝑛where𝑣¯=𝐓𝑢¯
> >
> > 1.1.
> > I believe, n is the number of generalized coordinates for the ny bodies,
> > as you mention below, below eq(185), but would it be a good idea to
> mention
> > it here, where you first introduce it?
> >
> > 1.2.
> > I assume that J_subscript(v_bar), subscript(q_bar) means
> > v_bar.Jacobian(q_bar)
> >
> > If I am right, should it not be J_subscript(v_bar), subscript(*u_bar*) ?
> >
> > At least, it seems that this is what you are doing here:
> >
> > T = v.jacobian(u)
> >
> >
> > 2. Eq(186, 187, 188, 189)
> >
> > Are these equations not the ‚heart‘ of the TMT Method?
> >
> > 3.
> >
> > You write:
> >
> > R_Bo = m*g*N.x + kl*q3*B.yR_Q = m/4*g*N.x - kl*q3*B.yT_A = -kt*q1*N.z +
> kt*q2*A.x
> >
> >
> > Of course, you have mentioned it before, but would it make sense to
> > mention that in
> >
> > R_Bo = m*g*N.x *+ kl*q3*B.y*
> >
> > the bold part is the ‚reactio force‘ on Bo from the particle Q?
> >
> > ( I missed this when I tried to model this pendulum with KanesMethod().
> It
> > should be obvious, but I failed to see it)
> >
> >
> > 4.
> > For my further education:
> >
> > Below eq(189), you write that these equations are equivalent to Kane‘s
> > Equations.
> > Is there an advantage to the TMT method?
> > It should be, as it was developed 35 years later.
> >
> >
> >
> >
> > On Fri 10. Jun 2022 at 18:25 Peter Stahlecker ***@***.***>
> > wrote:
> >
> >> Dear Jason,
> >>
> >> Clear, I missed the multiplication, did not realize that Mv.diff(t)
> meant
> >> M*v.diff(t).
> >> Thanks!
> >>
> >> Peter
> >>
> >> On Fri 10. Jun 2022 at 10:30, Jason K. Moore ***@***.***>
> >> wrote:
> >>
> >>> Peter,
> >>>
> >>> The difference is mathematical. $\mathbf{M}\frac{d\bar{v}}{dt}$ is
> >>> different than $\frac{d\mathbf{M}\bar{v}}{dt}$ if $\mathbf{M}$ is a
> >>> function of time, which it is in my presentation where I express the
> >>> inertias in the N reference frame, making them functions of the
> coordinates.
> >>>
> >>> —
> >>> Reply to this email directly, view it on GitHub
> >>> <
> https://github.com/moorepants/learn-multibody-dynamics/pull/80#issuecomment-1151942526
> >,
> >>> or unsubscribe
> >>> <
> https://github.com/notifications/unsubscribe-auth/AT5MQUTOKVHEVA7U5TYIR7TVOLDVLANCNFSM5YAII5XA
> >
> >>> .
> >>> You are receiving this because you were mentioned.Message ID:
> >>> ***@***.***>
> >>>
> >> --
> >> Best regards,
> >>
> >> Peter Stahlecker
> >>
> > --
> > Best regards,
> >
> > Peter Stahlecker
> >
> --
> Best regards,
>
> Peter Stahlecker
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/moorepants/learn-multibody-dynamics/pull/80#issuecomment-1152841277>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACDMJ4E6OGABS7ONR4ZTU3VOP5ZJANCNFSM5YAII5XA>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>

Peter230655 commented:
Dear Jason,

Thanks! Looking forward to it!

NB: I sent a few comments on your last lecture on TMT. Did you get them?

On Sat 11. Jun 2022 at 11:27, Jason K. Moore ***@***.***>
wrote:

> Peter,
>
> Yes, for sure. It will probably be sometime later this summer. I have to
> fix a bunch of things for the PDF to build correctly.
>
> Jason
> moorepants.info
> +01 530-601-9791
>
>
> On Sat, Jun 11, 2022 at 4:55 AM Peter230655 ***@***.***>
> wrote:
>
> > Dear Jason,
> >
> > You wrote to sympy:
> >
> >
> >
> > *Once I get the PDF to build, I think it will amount to about 300 pages
> or
> > so. It is fairly complete and can mostly standalone*
> >
> > Once you have this PDF, can you send me a copy, or would whatever
> > proprietary rights prevent you from doing so?
> >
> > Thanks!
> >
> > Peter
> >
> > On Fri 10. Jun 2022 at 20:47 Peter Stahlecker ***@***.***>
> > wrote:
> >
> > > *Unconstrained Equations of Motion with the TMT Method*
> > >
> > > 1. Eq(180)
> > >
> > > You write:
> > >
> > > 𝐓=𝐉𝑣¯,𝑞¯∈ℝ6𝜈×𝑛where𝑣¯=𝐓𝑢¯
> > >
> > > 1.1.
> > > I believe, n is the number of generalized coordinates for the ny
> bodies,
> > > as you mention below, below eq(185), but would it be a good idea to
> > mention
> > > it here, where you first introduce it?
> > >
> > > 1.2.
> > > I assume that J_subscript(v_bar), subscript(q_bar) means
> > > v_bar.Jacobian(q_bar)
> > >
> > > If I am right, should it not be J_subscript(v_bar), subscript(*u_bar*)
> ?
> > >
> > > At least, it seems that this is what you are doing here:
> > >
> > > T = v.jacobian(u)
> > >
> > >
> > > 2. Eq(186, 187, 188, 189)
> > >
> > > Are these equations not the ‚heart‘ of the TMT Method?
> > >
> > > 3.
> > >
> > > You write:
> > >
> > > R_Bo = m*g*N.x + kl*q3*B.yR_Q = m/4*g*N.x - kl*q3*B.yT_A = -kt*q1*N.z +
> > kt*q2*A.x
> > >
> > >
> > > Of course, you have mentioned it before, but would it make sense to
> > > mention that in
> > >
> > > R_Bo = m*g*N.x *+ kl*q3*B.y*
> > >
> > > the bold part is the ‚reactio force‘ on Bo from the particle Q?
> > >
> > > ( I missed this when I tried to model this pendulum with KanesMethod().
> > It
> > > should be obvious, but I failed to see it)
> > >
> > >
> > > 4.
> > > For my further education:
> > >
> > > Below eq(189), you write that these equations are equivalent to Kane‘s
> > > Equations.
> > > Is there an advantage to the TMT method?
> > > It should be, as it was developed 35 years later.
> > >
> > >
> > >
> > >
> > > On Fri 10. Jun 2022 at 18:25 Peter Stahlecker ***@***.***>
> > > wrote:
> > >
> > >> Dear Jason,
> > >>
> > >> Clear, I missed the multiplication, did not realize that Mv.diff(t)
> > meant
> > >> M*v.diff(t).
> > >> Thanks!
> > >>
> > >> Peter
> > >>
> > >> On Fri 10. Jun 2022 at 10:30, Jason K. Moore ***@***.***>
> > >> wrote:
> > >>
> > >>> Peter,
> > >>>
> > >>> The difference is mathematical. $\mathbf{M}\frac{d\bar{v}}{dt}$ is
> > >>> different than $\frac{d\mathbf{M}\bar{v}}{dt}$ if $\mathbf{M}$ is a
> > >>> function of time, which it is in my presentation where I express the
> > >>> inertias in the N reference frame, making them functions of the
> > coordinates.
> > >>>
> > >>> —
> > >>> Reply to this email directly, view it on GitHub
> > >>> <
> >
> https://github.com/moorepants/learn-multibody-dynamics/pull/80#issuecomment-1151942526
> > >,
> > >>> or unsubscribe
> > >>> <
> >
> https://github.com/notifications/unsubscribe-auth/AT5MQUTOKVHEVA7U5TYIR7TVOLDVLANCNFSM5YAII5XA
> > >
> > >>> .
> > >>> You are receiving this because you were mentioned.Message ID:
> > >>> ***@***.***>
> > >>>
> > >> --
> > >> Best regards,
> > >>
> > >> Peter Stahlecker
> > >>
> > > --
> > > Best regards,
> > >
> > > Peter Stahlecker
> > >
> > --
> > Best regards,
> >
> > Peter Stahlecker
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/moorepants/learn-multibody-dynamics/pull/80#issuecomment-1152841277
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AACDMJ4E6OGABS7ONR4ZTU3VOP5ZJANCNFSM5YAII5XA
> >
> > .
> > You are receiving this because you modified the open/close state.Message
> > ID: ***@***.***>
> >
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/moorepants/learn-multibody-dynamics/pull/80#issuecomment-1152863228>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUTDCHBOVNJWBEFLULLVOQTD5ANCNFSM5YAII5XA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
Best regards,

Peter Stahlecker

moorepants commented:
> NB: I sent a few comments on your last lecture on TMT. Did you get them?

No, I've had some others say emails are bouncing. Not sure what is going on. moorepants@gmail.com should work if my tudelft is bouncing.

moorepants commented:
> Unconstrained Equations of Motion with the TMT Method
> 1. Eq(180)
> You write:
> 𝐓=𝐉𝑣¯,𝑞¯∈ℝ6𝜈×𝑛where𝑣¯=𝐓𝑢¯
> 1.1.
> I believe, n is the number of generalized coordinates for the ny bodies, as you mention below, below eq(185), but would it be a good idea to mention it here, where you first introduce it?

Done.

> 1.2.
> I assume that J_subscript(v_bar), subscript(q_bar) means
> v_bar.Jacobian(q_bar)
>
> If I am right, should it not be J_subscript(v_bar), subscript(u_bar) ?
>
> At least, it seems that this is what you are doing here:
>
> T = v.jacobian(u) 

Corrected and also added my Jacobian notation to the notation page.

> 2. Eq(186, 187, 188, 189)
> 
> Are these equations not the ‚heart‘ of the TMT Method?

Yes.

> 3.
> 
> You write:
> 
> `R_Bo = m*g*N.x + kl*q3*B.y`
> `R_Q = m/4*g*N.x - kl*q3*B.y`
> `T_A = -kt*q1*N.z + kt*q2*A.x`
> 
> Of course, you have mentioned it before, but would it make sense to mention that in
> 
> `R_Bo = m*g*N.x + kl*q3*B.y`
> 
> the bold part is the ‚reactio force‘ on Bo from the particle Q?
> 
> ( I missed this when I tried to model this pendulum with KanesMethod(). It should be obvious, but I failed to see it)

I added a reminder to include the reactions.

> 4.
> For my further education:
> 
> Below eq(189), you write that these equations are equivalent to Kane‘s Equations.
> Is there an advantage to the TMT method?
>  should be, as it was developed 35 years later.

I don't know if the inventors of the TMT method were intimately familiar with Kane's equations (or even the Gibbs-Apell equations which have similarities). The presentation of the TMT method in the book I cite only builds upon Newton, Euler, and Lagrange's work. TMT has some advantages. You can essentially write a vector of positions and angular velocities as a function of time in the inertial frame and with some simple matrix math you can get the equations of motion. So maybe it is preferable to some. It also let's you think of the system as single rigid bodies, which is a bit simpler and then transform it to the reduced form. I think the authors may be the first to write this `T^T*M&T` transformation down, which is a nice result. The vector notation Kane uses isn't as matrix-friendly, so this transformation is wrapped up in the for loops that I write for form Fr and Frstar. I am also not sure, because I haven't counted this, but it would be interesting to see which methods had fewer derivatives to take or fewer operations. I think the TMT method ends of up with nastier equations if you multiply everything out. Note that the use of trigsimp in my example problem at the end was required to bring Md and gd into readable form, whereas Kane's method will give you that form without needing much or any simplification. I think that I'll make some appendices in the future that show various EoM methods and compare them and list advantages and disadvantages. TMT, for example, requires Lagrange multipliers to handle the constraints (at least how the authors present it). For symbolic EoM generation, I think it boils down more to what you were taught and are comfortable with than large tradeoffs in the resulting method outputs. More recent methods (Featherstone and Jain) offer an improvement in computational complexity, they can form udot using recursive methods for open trees or bodies that is of computational complexity O(n) instead of O(n^3) for the LU decomposition linear system solve.

Peter230655 commented:
*Unconstrained Equations of Motion with the TMT Method*

1. Eq(180)

You write:

𝐓=𝐉𝑣¯,𝑞¯∈ℝ6𝜈×𝑛where𝑣¯=𝐓𝑢¯

1.1.
I believe, n is the number of generalized coordinates for the ny bodies, as
you mention below, below eq(185), but would it be a good idea to mention it
here, where you first introduce it?

1.2.
I assume that J_subscript(v_bar), subscript(q_bar) means
v_bar.Jacobian(q_bar)

If I am right, should it not be J_subscript(v_bar), subscript(*u_bar*) ?

At least, it seems that this is what you are doing here:

T = v.jacobian(u)


2. Eq(186, 187, 188, 189)

Are these equations not the ‚heart‘ of the TMT Method?

3.

You write:

R_Bo = m*g*N.x + kl*q3*B.yR_Q = m/4*g*N.x - kl*q3*B.yT_A = -kt*q1*N.z
+ kt*q2*A.x


Of course, you have mentioned it before, but would it make sense to mention
that in

R_Bo = m*g*N.x *+ kl*q3*B.y*

the bold part is the ‚reactio force‘ on Bo from the particle Q?

( I missed this when I tried to model this pendulum with KanesMethod(). It
should be obvious, but I failed to see it)


4.
For my further education:

Below eq(189), you write that these equations are equivalent to Kane‘s
Equations.
Is there an advantage to the TMT method?
It should be, as it was developed 35 years later.




On Fri 10. Jun 2022 at 18:25 Peter Stahlecker ***@***.***>
wrote:

> Dear Jason,
>
> Clear, I missed the multiplication, did not realize that Mv.diff(t) meant
> M*v.diff(t).
> Thanks!
>
> Peter
>
> On Fri 10. Jun 2022 at 10:30, Jason K. Moore ***@***.***>
> wrote:
>
>> Peter,
>>
>> The difference is mathematical. $\mathbf{M}\frac{d\bar{v}}{dt}$ is
>> different than $\frac{d\mathbf{M}\bar{v}}{dt}$ if $\mathbf{M}$ is a
>> function of time, which it is in my presentation where I express the
>> inertias in the N reference frame, making them functions of the coordinates.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/moorepants/learn-multibody-dynamics/pull/80#issuecomment-1151942526>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AT5MQUTOKVHEVA7U5TYIR7TVOLDVLANCNFSM5YAII5XA>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
> --
> Best regards,
>
> Peter Stahlecker
>
-- 
Best regards,

Peter Stahlecker

Peter230655 commented:
Dear Jason,

I enjoyed to be a small part of this!
Should you ever teach another class on mechanics, where you think I amy be
of use, please let me know.
I will be happy to assist!

Take care!

Peter

On Thu 9. Jun 2022 at 11:31 Jason K. Moore ***@***.***> wrote:

> BTW, this is the last chapter for the purposes of the class! Hooray! Nice
> accomplishment. You two have been a great help. I very much appreciate it :)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/moorepants/learn-multibody-dynamics/pull/80#issuecomment-1150705195>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUUCCHRQDMBONGAWUY3VOGCCNANCNFSM5YAII5XA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
Best regards,

Peter Stahlecker
